servicex/rabbit_adaptor.py

```diff
@@ -169,25 +174,35 @@ def basic_publish(self, exchange, routing_key, body):
         while True:
             try:
                 channel = self.channel
+
                 channel.basic_publish(exchange=exchange,
                                       routing_key=routing_key,
-                                      body=body)
+                                      body=body,
+                                      properties=pika.BasicProperties(delivery_mode=1),
+                                      mandatory=True)
                 return
+
             except pika.exceptions.ConnectionClosedByBroker:
                 # Uncomment this to make the example not attempt recovery
                 # from server-initiated connection closure, including
                 # when the node is stopped cleanly
                 #
                 # break
                 continue
+            except pika.exceptions.ChannelWrongStateError:
+                self.reset_closed()
+                current_app.loggger.info("Channel in wrong state. Reset and see if that fixes it")
```

Comment:
This will make the message "reset and see..." come after the reset. Perhaps exchange the order of messages? Otherwise, change the text of the message. I'm harping on this because I've seen error messages in servicex that say "I'm going to do x" and the log lines from doing x appear above.

Reply:
Good point. Fixed this and the others in this class